tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.cc

```diff
@@ -25,6 +25,18 @@ limitations under the License.
 namespace tensorflow {
 namespace tensorrt {
 
+void SetImplicitBatchLikeProfile(const std::vector<nvinfer1::Dims>& dimvec,
```

Comment:
How about this:
   (1) rename the routine to GetProfileDimensions
   (2) make the routine take an additional parameter, ProfileStrategy
   (3) replace the switch on ProfileStrategy in TrtShapeOptimizationProfile::InitProfiles() with a call to this routine